internal/service/autoscaling/policy.go

```diff
@@ -87,6 +87,276 @@ func ResourcePolicy() *schema.Resource {
 							MaxItems: 1,
 							Elem: &schema.Resource{
 								Schema: map[string]*schema.Schema{
+									"customized_scaling_metric_specification": {
+										Type:          schema.TypeList,
+										Optional:      true,
+										MaxItems:      1,
+										ConflictsWith: []string{"predictive_scaling_configuration.0.metric_specification.0.predefined_scaling_metric_specification"},
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"metric_data_queries": {
+													Type:     schema.TypeList,
+													Required: true,
+													MaxItems: 10,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"id": {
+																Type:         schema.TypeString,
+																Required:     true,
+																ValidateFunc: validation.StringLenBetween(1, 255),
+															},
+															"expression": {
```

Comment:
From https://docs.aws.amazon.com/autoscaling/ec2/APIReference/API_MetricDataQuery.html it mentions `Within each MetricDataQuery object, you must specify either Expression or MetricStat, but not both.` Hence, we will need a `ConflictsWith` specified here

Reply:
Good catch, will add.

Reply:
Actually, I ran into issues while trying to implement this. The `metric_data_queries` is a list, while it seemed there is some issues using `ConflictWith` with [List/Set](https://github.com/hashicorp/terraform-plugin-sdk/issues/71). Any thought on this ?

Reply:
Conclusion from this discussion was to write that the argument conflict in the documentation. This problem is different because the length of this List can be > 1. There are other patterns where the length of the list is 1 and it is possible to index the nested attributes with `.0`, for example:

https://github.com/hashicorp/terraform-provider-aws/blob/bb560f77070a5b5aada8acd1d58df7a8b25c1f71/internal/service/emr/cluster.go#L125-L152